sklearn/metrics/_classification.py

```diff
@@ -2285,6 +2285,48 @@ def log_loss(y_true, y_pred, *, eps=1e-15, normalize=True, sample_weight=None,
     return _weighted_sum(loss, sample_weight, normalize)
 
 
+def _check_valid_multiclass_decision_shape(y_true_unique, pred_decision,
```

Comment:
I would be okay with creating another function if we can use this check in other metrics. I can be missing one, but I do not see a metric that does this validation.